### PR TITLE
Aliasing

### PR DESCRIPTION
**Pull request by vanceza**
Switched over from having only the concept of "nodes", to having both "nodes" and "instances". No other changes, aliasing is not actually added yet.

An instance is a client-side-only concept, it is never written to the datastore. lastViewRoot does contain information about an instance, but the datastore now treats this as an opaque token with no default value.

An instance looks like

    {
      id: <node-id>,
      crumbs: { 
        parent: <parent-node-id>,
        crumbs: <further-ancestors>
      }
    }

For example,

    {
     id: 3,
      crumbs: { 
        parent: 2,
        crumbs: { parent: 1, crumbs: { parent: 0 } }
      }
    }

### CODE REVIEW

**Comment by vanceza** [comment]
(Working on merging the new marks)

**Comment by vanceza** [comment]
Latest changes are merged

File: assets/js/keyBindings.coffee [hunk 1]
```diff
@@ -773,6 +776,12 @@ if module?
           finishes_visual: true
           fn: (cursor, options = {}) ->
             @view.yankBetween @view.cursor, cursor, options
+        CLONE:
+          display: 'Yank blocks as a clone'
+          drop: true # TODO: Is this correct? Re-check after Jeff documents
```
Comment (WuTheFWasThat) [inline assets/js/keyBindings.coffee]: yeah drop: true is fine, it means the sequence won't be saved for things like the repeat operator, which i think makes sense here

**Comment by WuTheFWasThat** [comment]
needs tests!  still don't like the way crumbs work, but that can be changed

**Comment by vanceza** [comment]
Yeah, when I made the PR it was just for data structure changes, and that's what you should merge if anything (which doesn't need more tests).

Aliasing is not done and doesn't have tests because it's not working. I'll add them.

**Pull request closed without merging**